src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs

```diff
@@ -99,6 +104,13 @@ public static void ReportNonFatal(Exception exception)
                     exceptionObject: exception,
                     gatherEventDetails: faultUtility =>
                     {
+                        if (forceDump)
+                        {
+                            // Let's just send a maximum of three; number chosen arbitrarily
+                            if (Interlocked.Increment(ref s_dumpsSubmitted) <= 3)
```

Comment:
This will be dumps per session of VS. Honestly, we could get away with 1 here and I'd still be happy

Reply:
I'm not sure what triggers some of these faults but in some cases in unit testing we've seen users with sessions that last for a day or more (where they could run into multiple faults) are more likely to repro certain edge cases. So that number in code seems alright.

Reply:
In the bug I'm trying to investigate, I absolutely want a limit because once the case hits it might be hitting a bunch. I expect the functional mitigation (the try/catch) will help that but I don't want to bet on it.

@ryzngard: my concern with "1" was whether we'd be filtering a bit _too_ much; I don't have a great sense yet how many sessions my bug is hitting -- I  just haven't done the digging yet -- so was a little worried that too few dumps might be bad there.

Reply:
I'm not picky on the number here, just pointing out since this is a static it gets reset when VS closes. I don't think 3 is obscene. We could also map number captured to level as discussed in https://github.com/dotnet/roslyn/pull/57680/files/3acf384b4808c55137ddb2390634ed3ff91a58b7#r746999647